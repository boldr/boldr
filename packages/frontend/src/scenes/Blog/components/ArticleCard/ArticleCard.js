/* @flow */
import React from 'react';
import { format } from 'date-fns';
import Link from 'react-router-dom/Link';
import classnames from 'classnames';
import { Button, Card, CardTitle, CardText, CardActions, Media, StyleClasses } from '@boldr/ui';
import type { Article } from '../../../../types/boldr';

import TagBlock from '../TagBlock';

const BASE_ELEMENT = StyleClasses.ARTICLE_CARD;

type Props = {
  className?: string,
  article: Article,
};

const ArticleCard = (props: Props) => {
  const formattedDate = format(props.article.createdAt, 'MM/DD/YYYY');
  const classes = classnames(BASE_ELEMENT, props.className);
  const { title, image, slug, tags, excerpt } = props.article;
  return (
    <div className={classes}>
      <Card style={{ maxWidth: 600 }} className="md-block-centered">
        <Media>
          <img src={image} alt={`${title} feature image`} role="presentation" />
        </Media>
        <CardTitle title={title} subtitle={formattedDate} />
        <CardActions expander>
          <Link to={`/blog/${slug}`}>
            <Button kind="primary" outline>
              Read More
            </Button>
          </Link>
        </CardActions>
        <CardText expandable>{excerpt}</CardText>
        <TagBlock tags={tags} />
      </Card>
    </div>
  );
};

export default ArticleCard;
