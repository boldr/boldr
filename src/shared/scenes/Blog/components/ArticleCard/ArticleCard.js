/* @flow */
import React from 'react';
import Link from 'react-router-dom/Link';
import { connect } from 'react-redux';
import dateFns from 'date-fns';
import classnames from 'classnames';
import { StyleClasses } from '../../../../theme/styleClasses';
import Col from '~components/Layout/Col';
import Row from '~components/Layout/Row';
import Button from '~components/Button';
import Card, {
  CardHeader,
  CardMedia,
  CardContent,
  CardActions,
} from 'material-ui/Card';
import { selectArticle } from '../../state/articles/actions';

import TagBlock from '../TagBlock';

const BASE_ELEMENT = StyleClasses.POST_CARD;
type Props = {
  className: string,
  article: Article,
  dispatch: () => void,
};

export const ArticleCard = (props: Props) => {
  const formattedDate = dateFns.format(props.article.createdAt, 'MM/DD/YYYY');
  const classes = classnames(BASE_ELEMENT, props.className);

  function transitionPost() {
    const { article } = props;
    props.dispatch(selectArticle(article));
  }
  const { title, featureImage, slug, tags, excerpt } = props.article;
  return (
    <div className={classes}>
      <Card>
        <CardHeader title={title} subheader={formattedDate} />
        <CardMedia>
          <img src={featureImage} alt={`${title} feature image`} />
        </CardMedia>

        <CardContent>
          {excerpt}
          <Row>
            <Col sm={12}>
              <Link to={`/blog/${slug}`} className="readmore-link">
                <Button kind="primary" onClick={transitionPost} outline>
                  Read More
                </Button>
              </Link>
            </Col>
          </Row>
        </CardContent>
        <CardActions>
          <TagBlock tags={tags} />
        </CardActions>
      </Card>
    </div>
  );
};

export default connect(state => state, null, null, { pure: true })(ArticleCard);
