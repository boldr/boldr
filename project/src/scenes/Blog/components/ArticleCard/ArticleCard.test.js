import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { shallowToJson } from 'enzyme-to-json';
import ArticleCard from './ArticleCard';

const articleData = {
  id: 'c265ed52-b9c4-42e8-9b46-7bc409bfec48',
  featured: true,
  slug: 'this-test-slug',
  title: 'This Test Slug',
  image: 'https://boldr.io/logo.png',
  createdAt: '2017-09-11 04:33:36.524+00',
  excerpt: 'Abcdc',
  rawContent: {},
  tags: [{ id: 'c265ed52-b9c4-42e8-9b46-7bc409bfec33', name: 'Tag' }],
  author: {
    id: 'c265ed52-b9c4-42e8-9b46-7bc409bfec89',
    email: 'admin@boldr.io',
    profile: {
      username: 'User',
      firstName: 'Jobe',
      lastName: 'John',
      avatarUrl: 'https://boldr.io/logo.png',
    },
  },
};

describe('<ArticleCard />', () => {
  // it('accepts props and renders them.', () => {
  //   const wrapper = shallow(
  //     <ArticleCard
  //       id={articleData.id}
  //       image={articleData.image}
  //       title={articleData.title}
  //       excerpt={articleData.excerpt}
  //       slug={articleData.slug}
  //       author={articleData.author}
  //       tags={articleData.tags}
  //     />,
  //   );
  //   expect(wrapper.instance().props.image).toEqual('https://boldr.io/logo.png');
  //   expect(wrapper.instance().props.title).toEqual('This Test Slug');
  //   expect(wrapper.instance().props.excerpt).toEqual('Abcdc');
  //   expect(wrapper.instance().props.slug).toEqual('this-test-slug');
  // });
  it('matches the snapshot', () => {
    const wrapper = shallow(<ArticleCard {...articleData} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
