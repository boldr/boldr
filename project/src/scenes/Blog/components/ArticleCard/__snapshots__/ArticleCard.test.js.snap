// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleCard /> matches the snapshot 1`] = `
<ArticleCard__Article
  featuring={true}
  key="c265ed52-b9c4-42e8-9b46-7bc409bfec48"
>
  <ArticleCard__ImageLink
    to="/blog/this-test-slug"
  >
    <ArticleCard__Image
      image="https://boldr.io/logo.png"
    />
  </ArticleCard__ImageLink>
  <ArticleCard__Content>
    <ArticleCard__ContentLink
      to="/blog/this-test-slug"
    >
      <ArticleCard__ArticleHeader>
        <ArticleCard__Title>
          This Test Slug
        </ArticleCard__Title>
        09/10/2017
      </ArticleCard__ArticleHeader>
      <ArticleCard__Excerpt>
        <p>
          Abcdc
        </p>
      </ArticleCard__Excerpt>
    </ArticleCard__ContentLink>
    <ArticleCard__Footer>
      <ArticleCard__AuthorLink
        to="/profiles/User"
      >
        <ArticleCard__AuthorImage
          alt="Jobe"
          src="https://boldr.io/logo.png"
        />
        <ArticleCard__Author>
          Jobe
        </ArticleCard__Author>
      </ArticleCard__AuthorLink>
      <TagBlock
        tags={
          Array [
            Object {
              "id": "c265ed52-b9c4-42e8-9b46-7bc409bfec33",
              "name": "Tag",
            },
          ]
        }
      />
    </ArticleCard__Footer>
  </ArticleCard__Content>
</ArticleCard__Article>
`;
